zebra-chain/src/serialization.rs

```diff
@@ -299,6 +306,23 @@ pub trait ReadZcashExt: io::Read {
 /// Mark all types implementing `Read` as implementing the extension.
 impl<R: io::Read + ?Sized> ReadZcashExt for R {}
 
+/// Helper for deserializing more more succinctly via type inference
+pub trait DeserializeInto {
```

Comment:
All these Deserialize's should be ZcashDeserialize because zcash(de)serialization is about the format on the wire that consensus depends on.